posts/views.py

```diff
@@ -16,12 +17,150 @@ def post_create(request):
     if request.method == 'POST':
         post_form = PostForm(request.POST, request.FILES)
         if post_form.is_valid():
-            post_form.save(commit=True)
+            post = post_form.save(commit=False)
+            post.owner = request.user
+            post.save()
+            for choice in post.choices:
```

Comment:
You can get ingredients or ingredient ids from the user. You should update this part after removing `choices`.